modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/OperationsStateHandler.scala

```diff
@@ -46,15 +52,27 @@ class OperationsStateHandler[M](modelRW: ModelRW[M, SequencesOnDisplay])
 
     case RequestResourceRun(id, _, r) =>
       // TODO Remove this when the response comes over the WS channel
+      logger.info(s"*** OperationsStateHandler handleRequestOperation RequestResourceRun: r=$r")
       updated(
         value.markOperations(
           id,
           TabOperations
             .resourceRun(r)
             .set(ResourceRunOperation.ResourceRunInFlight.some)))
 
-    case RunResource(id, _, r) =>
-      updated(value.markOperations(id, TabOperations.resourceRun(r).set(none)))
+//    case RunResource(id, _, r) =>
+//      logger.info(s"*** OperationsStateHandler handleRequestOperation RunResource: r=$r, r set none = ${TabOperations.resourceRun(r).set(none)}")
+//      updated(value.markOperations(id, TabOperations.resourceRun(r).set(none)))
+  }
+
+  def singleRunCompletedMessage: PartialFunction[Any, ActionResult[M]] = {
+    case ServerMessage(SingleActionEvent(SingleActionOp.Completed(id, r))) =>
+      logger.info(s"*** OperationsStateHandler singleRunCompletedMessage: r=$r")
+//      val v = value.markOperations(id, TabOperations.resourceRun(r).set(none))
+//      val f = Future(v)
+//      val e = Effect(f)
+    updated(value.markOperations(id, TabOperations.resourceRun(r).set(none))
```

Comment:
This is what I would like to trigger when the configuration completes.

I'm not sure if this is the correct way to update the button (I discussed alternatives with @cquiroz but had trouble getting them to work due to missing information - e.g. step number - or implicits) so I used the same technique as in the commented out `RunResource` case, which did set the button back to the non-running state.

Reply:
Right, the issue is that each `ActionHandler` is processin some of the events. In the case of events from the server they are handled in `ServerMessagehandler`. So `singleRuncompleted` is never called

Reply:
This is somewhat explained at
https://diode.suzaku.io/usage/Handlers.html

Reply:
Hmmm... I know you recommended putting this in `ServerMessageHandler`, but I don't think I have the step number information available here, which I seem to need, I think?

I'll try playing around with calls in there as per our earlier discussion, but I'm not sure how to get the missing information, and had some issue making the call with a missing implicit, which seemed weird.

Reply:
Yeah, checking now, and I don't have the step number available here, I don't think.

Reply:
Should I try seeing if I can pass it with `SingleActionOp`? That doesn't seem like it necessarily makes sense.

Reply:
Yes

Reply:
Never mind... I didn't refresh and see your comments. Okay, I'll see if I can add these to the event.